compiler/passes/src/static_single_assignment/expression.rs

```diff
@@ -44,15 +44,26 @@ use leo_span::{Symbol, sym};
 
 use indexmap::IndexMap;
 
+impl SsaFormingVisitor<'_> {
+    /// Consume this expression and assign it to a variable (unless it's already an Identifier).
+    pub fn consume_expression_and_define(&mut self, input: Expression) -> (Expression, Vec<Statement>) {
```

Comment:
To confirm, the way `consume_expression_and_define` is used in this pass changes the AST produced by SSA by not returning an intermediate variable for already simple binary expressions.